### PR TITLE
[ISSUE #8] Add fc sink task

### PR DESCRIPTION
**Pull request by zhaohai1299002788**
add fc sink task

### CODE REVIEW

**Comment by zhouxinyu** [comment]
It seems that we should move the directory to `connectors` instead of the root?

**Comment by zhaohai1299002788** [comment]
> 看来我们应该将目录移动到`connectors`而不是根目录？

Modified engineering structure

File: aliyun/rocketmq-connect-fc/src/main/resources/log4j.properties [hunk 1]
```diff
@@ -0,0 +1,28 @@
+log4j.rootCategory=INFO, console
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.target=System.err
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yy/MM/dd HH:mm:ss} %p %c{1}: %m%n
+
+# Set the default spark-shell log level to WARN. When running the spark-shell, the
+# log level for this class is used to overwrite the root logger's log level, so that
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply aliyun/rocketmq-connect-fc/src/main/resources/log4j.properties]: Irrelevant information deleted

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkConnector.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package com.aliyun.rocketmq.connect.fc.sink;
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkConnector.java:1]: change package to org.apache.rocketmq.connect.xxx

File: connectors/aliyun/rocketmq-connect-fc/pom.xml [hunk 3]
```diff
@@ -0,0 +1,211 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.aliyun</groupId>
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-fc/pom.xml:7]: change groupId  to   org.apache.rocketmq

File: connectors/aliyun/rocketmq-connect-fc/pom.xml [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <slf4j.version>1.7.7</slf4j.version>
+        <logback.version>1.0.13</logback.version>
+        <openmessaging-connector.version>0.1.2-SNAPSHOT</openmessaging-connector.version>
+        <junit.version>4.11</junit.version>
+        <assertj.version>2.6.0</assertj.version>
+        <mockito.version>3.2.4</mockito.version>
+        <openmessaging-connector.version>0.1.2-SNAPSHOT</openmessaging-connector.version>
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-fc/pom.xml:21]: <openmessaging-connector.version>0.1.2-SNAPSHOT</openmessaging-connector.version>

duplicate

File: connectors/aliyun/rocketmq-connect-fc/src/main/resources/logback.xml [hunk 5]
```diff
@@ -0,0 +1,90 @@
+<configuration>
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/src/main/resources/logback.xml]: delete logback

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkConnector.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Override
+    public void init(KeyValue config) {
+        region = config.getString(FcConstant.REGION_CONSTANT);
+        accessKey = config.getString(FcConstant.ACCESS_KEY_CONSTANT);
+        accessSecretKey = config.getString(FcConstant.ACCESS_SECRET_KEY_CONSTANT);
+        accountId = config.getString(FcConstant.ACCOUNT_ID_CONSTANT);
+        serviceName = config.getString(FcConstant.SERVICE_NAME_CONSTANT);
+        functionName = config.getString(FcConstant.FUNCTION_NAME_CONSTANT);
+        invocationType = config.getString(FcConstant.INVOCATION_TYPE_CONSTANT, null);
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkConnector.java:74]: delete duplicate code

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkTask.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    public void validate(KeyValue config) {
+        if (StringUtils.isBlank(config.getString(FcConstant.REGION_CONSTANT))
+            || StringUtils.isBlank(config.getString(FcConstant.ACCESS_KEY_CONSTANT))
+            || StringUtils.isBlank(config.getString(FcConstant.ACCESS_SECRET_KEY_CONSTANT))
+            || StringUtils.isBlank(config.getString(FcConstant.ACCOUNT_ID_CONSTANT))
+            || StringUtils.isBlank(config.getString(FcConstant.SERVICE_NAME_CONSTANT))
+            || StringUtils.isBlank(config.getString(FcConstant.FUNCTION_NAME_CONSTANT))) {
+            throw new RuntimeException("fc required parameter is null !");
+        }
+        try {
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkTask.java:91]: Added queue name check, and checked the legitimacy of access key and secret key.

File: connectors/aliyun/rocketmq-connect-fc/pom.xml (hunk 4 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/pom.xml:20]: Processed

File: connectors/aliyun/rocketmq-connect-fc/pom.xml (hunk 3 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/pom.xml]: Processed

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkConnector.java (hunk 2 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/src/main/java/com/aliyun/rocketmq/connect/fc/sink/FcSinkConnector.java]: Processed

File: connectors/aliyun/rocketmq-connect-fc/README.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+|------------------------|---------|----------------|----------------------------------|--|
+|region                  | String  | YES            | 地域                               | cn-hangzhou|
+|accessKey               | String  | YES            | 阿里云身份验证，在阿里云用户信息管理控制台获取                    | xxxx |
+|accessSecretKey         | String  | YES            | 阿里云身份验证，在阿里云用户信息管理控制台获取                     | xxx |
+|accountId               | String  | YES            | 阿里云yourAccountId                      | xxxx |
+|serviceName             | String  | YES            | 服务名称 | xxxx |
+|functionName            | String  | YES            | 函数名称 | xxxx |
+|invocationType    | String | NO             | 同步或者异步                           | null |
+|qualifier        | String | NO             | 服务版本和别名                          | LATEST |
+
```
[the remark this answers is not in the document]
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/README.md:49]: Added trasform configuration.

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+import io.openmessaging.internal.DefaultKeyValue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FcSinkConnector extends SinkConnector {
+
+    private String region;
+
+    private String accessKey;
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java:16]: 使用官方名称：accessKeyId 和accessKeySecret

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+import io.openmessaging.connector.api.component.task.Task;
+import io.openmessaging.connector.api.component.task.sink.SinkConnector;
+import io.openmessaging.internal.DefaultKeyValue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FcSinkConnector extends SinkConnector {
+
+    private String region;
```
Comment (2011shenlin) [inline connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java:14]: 改用RegionId

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java (hunk 9 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java]: 已处理

File: connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java (hunk 10 above)
Comment (zhaohai1299002788) [reply connectors/aliyun/rocketmq-connect-fc/src/main/java/org/apache/rocketmq/connect/fc/sink/FcSinkConnector.java]: 已处理

**Pull request merged**